homeassistant/components/efergy/__init__.py

```diff
@@ -16,37 +16,35 @@
 from .const import DEFAULT_NAME, DOMAIN
 
 PLATFORMS = [Platform.SENSOR]
+type EfergyConfigEntry = ConfigEntry[Efergy]
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: EfergyConfigEntry) -> bool:
     """Set up Efergy from a config entry."""
-    api = Efergy(
+    entry.runtime_data = Efergy(
         entry.data[CONF_API_KEY],
         session=async_get_clientsession(hass),
         utc_offset=hass.config.time_zone,
         currency=hass.config.currency,
     )
```

Comment:
We usually don't store the data until after the integration has had a chance to raise `ConfigEntryNotReady`/`ConfigEntryAuthFailed`